lldb/include/lldb/Target/ThreadPlanStack.h

```diff
@@ -128,15 +133,34 @@ class ThreadPlanStackMap {
 
   void AddThread(Thread &thread) {
     lldb::tid_t tid = thread.GetID();
-    m_plans_list.emplace(tid, thread);
+    std::shared_ptr<ThreadPlanStack> new_elem_sp(new ThreadPlanStack(thread));
+    m_plans_sp_container.push_back(new_elem_sp);
+    m_plans_list.emplace(tid, new_elem_sp.get());
   }
 
   bool RemoveTID(lldb::tid_t tid) {
     auto result = m_plans_list.find(tid);
     if (result == m_plans_list.end())
       return false;
-    result->second.ThreadDestroyed(nullptr);
+    ThreadPlanStack *removed_stack = result->second;
     m_plans_list.erase(result);
+    // Also remove this from the stack storage:
+    PlansStore::iterator end 
+        = m_plans_sp_container.end();
+    PlansStore::iterator iter; 
+    for (iter = m_plans_sp_container.begin();
+         iter != end; iter++) {
```

Comment:
Is this one of those times where we actually do need to use pre-increment on the iterator.